mewarpx/mewarpx/assemblies.py

```diff
@@ -316,6 +316,95 @@ def __init__(self, V, T, WF, read_scraped_particles=True):
         self.scraper_label = 'z_lo'
 
 
+class PatchyCathode(object):
+    """A basic wrapper to define a semi-infinite plane for a patchy cathode.
+    Special aspects of the patchy cathode involve instantiation of two assembly
+    objects for the low and high work-function patches of the cathode and
+    setting a spatially varying boundary potential. The spatially varying
+    boundary potential is simplified by using an embedded boundary rather than
+    the domain boundary. Simplified assemblies similar to ``Rectangle`` is used.
+
+    Arguments:
+        V (float): Cathode fermi-level - phi_bar (far field vacuum level).
+        T (float): Temperature of the cathode in Kelvin.
+        phi_bar (float): Average work-function.
+        delta_phi (float): Difference between max and min work-function values.
+        patch_size (float): Patch size in meters.
+    """
+    geoms = ['XZ']
+
+    def __init__(self, V, T, phi_bar, delta_phi, patch_size,
+                 read_scraped_particles=True):
+
+        self.patch_size = patch_size
+        self.delta_phi = delta_phi
+
+        # the patchy cathode extends 1.5 cell width into the domain
+        # if the boundary falls on a cell edge, the calculation fails
+        self.z_max = mwxrun.zmin + 0.5*mwxrun.dz
+        self.z_min = mwxrun.zmin - 0.5*mwxrun.dz
+
+        # build the two patch segments of the cathode
+        self.high_wf_patch = self.PatchSet(
+            V-delta_phi/2.0, T=T, WF=phi_bar+delta_phi/2.0,
+            name='cathode_high_wf', patch_size=patch_size, x_start=0.0,
+            z=self.z_max
+        )
+        self.low_wf_patch = self.PatchSet(
+            V+delta_phi/2.0, T=T, WF=phi_bar-delta_phi/2.0,
+            name='cathode_low_wf', patch_size=patch_size, x_start=patch_size,
+            z=self.z_max
+        )
+        self.cathode_list = [self.high_wf_patch, self.low_wf_patch]
+
+        # Negative means outside of object
+        self.implicit_function = (
+            f"-max(max(x-{mwxrun.xmax},{mwxrun.xmin}-x),"
+            f"max(z-{self.z_max},{self.z_min}-z))"
+        )
+        self.scraper_label = 'eb'
+        # set the boundary potential - note that this expression assumes
+        # that the higher WF patch starts at x=0 (continuing to the right)
+        self.V = (f"({V}+"
+            f"{delta_phi}*(fmod(ceil((abs(x/{patch_size}+0.5)+0.5)),2)-0.5))"
+        )
+        Assembly._install_in_simulation(self)
```

Comment:
Please add comments here (or in the docstring if you prefer) for why this is needed.

Reply:
You mean the call to `_install_in_simulation`, right?

Reply:
yep thanks!